src/validation.h

```diff
@@ -463,6 +463,12 @@ void UpdateUncommittedBlockStructures(CBlock& block, const CBlockIndex* pindexPr
 /** Produce the necessary coinbase commitment for a block (modifies the hash, don't call for mined blocks). */
 std::vector<unsigned char> GenerateCoinbaseCommitment(CBlock& block, const CBlockIndex* pindexPrev, const Consensus::Params& consensusParams);
 
+/** Produce the SCDB hashMerkleRoot coinbase commitment for a block */
+CScript GenerateSCDBCoinbaseCommitment();
+
+/** Produce a BMM h* coinbase commitment for a block */
+CScript GeneratBMMCriticalHashCommitment();
```

Comment:
I think these should probably be pure functions, i.e. they take the commitment hash as a parameter to the function -- not making a call in the middle of the function to retrieve the hash. This makes it easier to test/reason about. 

For instance the caller of this function will need to retrieve a hash from the miner's `OP_BV output wallet` and then call this to create the actual `CScript`. This should probably all happen in `src/rpc/mining.cpp` and I was trying to convey this on my pull request: 

https://github.com/drivechain-project/bitcoin/pull/13/files#diff-9651347c8e00bed3ddc7631de569406dR549

Reply:
This is something I wanted to talk about on during the IRC meeting today. It seems like a good idea to do things like BIP 141 but maybe putting everything into validation isn't actually necessary?

Reply:
Is this function you are trying to mimic? It seems pseudo stateless, as it takes as input all of the information it needs to calculate the witness commitment, but modifies the block given as a parameter instead of returning a copy of the block with the new witness commitment included in the coinbase tx:

https://github.com/drivechain-project/bitcoin/blob/mainchainBMM/src/validation.h#L464

Reply:
Yes I'm trying to copy BIP 141 as much as possible. We should update as you suggested, do you think we should keep it in validation though or just put it in one of the miner files?